compiler/infra/BitVector.hpp

```diff
@@ -838,6 +874,40 @@ class TR_BitVector
       }
    #endif
 
+   /**
+    * @brief Computes number of bytes required for serializing this object
+    *
+    * @return Number of bytes required for serializing this object
+    */
+   uint32_t getSizeForSerialization() const
+      {
+      uint32_t size = sizeof(TR_SerializedBitVector);
+      if (_numChunks > 0)
+         {
+         size += (_numChunks * sizeof(*_chunks));
+         }
+      return size;
+      }
+
+   /**
+    * @brief Serialize this object
+    * @param [in] serializer used for serializing this object
+    */
+   void serialize(uint8_t * &buffer) const
```

Comment:
Is this a standardized signature across OMR? I'm not quite understanding why this is not a `void*` given we cast the type anyway.

Reply:
@fjeremic `uint8_t *` makes it easier to increment the buffer and I want the buffer to be updated which is why I made it a reference. But I can instead use the following signature:
`void *serialize(void *buffer) const`
This would return the updated buffer. Does that look ok?

Reply:
No this is fine. I think we should at the very least document that `buffer` will get mutated as data is written and that it is up to the user to determine that there is enough space available.

Reply:
@fjeremic Updated the comment as suggested.